comment/validators.py

```diff
@@ -0,0 +1,101 @@
+
+from comment.models import Comment
+
+
+class CommentBadRequest(APIException):
+    status_code = status.HTTP_400_BAD_REQUEST
+    default_detail = 'Bad Request'
+
+    def __init__(self, detail=None, status_code=None):
+        if status_code:
+            self.status_code = status_code
+        if not detail:
+            detail = self.default_detail
+        self.detail = detail
+
+
+class BaseValidatorMixin:
+    api = False
+    app_name = None
+    model_name = None
+    model_id = None
+    parent_id = None
+    error = None
+
+    def dispatch(self, request, *args, **kwargs):
+        """
+            let rest framework handle the exception to choose the right renderer
+            validate method shall be called in the derived API class
+        """
+        if self.api:
+            return super().dispatch(request, *args, **kwargs)
+        try:
+            self.validate(request)
+        except CommentBadRequest as exc:
+            return JsonResponse({'type': _('error'), 'detail': _(exc.detail)}, status=400)
+        return super().dispatch(request, *args, **kwargs)
+
+    def validate(self, request):
+        self.app_name = request.GET.get("app_name") or request.POST.get("app_name")
+        self.model_name = request.GET.get("model_name") or request.POST.get("model_name")
+        self.model_id = request.GET.get("model_id") or request.POST.get("model_id")
+        self.parent_id = request.GET.get("parent_id") or request.POST.get("parent_id")
+
+
+class ContentTypeValidator(BaseValidatorMixin):
+    def validate(self, request):
+        super().validate(request)
+        if not self.model_name:
+            self.error = "model name must be provided"
+            raise CommentBadRequest(self.error)
+        if not self.model_id:
+            self.error = "model id must be provided"
+            raise CommentBadRequest(self.error)
+        if not self.app_name:
+            self.error = 'app name must be provided'
+            raise CommentBadRequest(self.error)
+
+        if not ContentType.objects.filter(app_label=self.app_name).exists():
+            self.error = f'{self.app_name} is NOT a valid app name'
+            raise CommentBadRequest(self.error)
+        try:
+            model_name = self.model_name.lower()
+            ct = ContentType.objects.get(model=model_name).model_class()
+            model_class = ct.objects.filter(id=self.model_id)
+            if not model_class.exists() and model_class.count() != 1:
+                self.error = f'{self.model_id} is NOT a valid model id for the model {self.model_name}'
+                raise CommentBadRequest(self.error)
+        except ContentType.DoesNotExist:
+            self.error = f'{self.model_name} is NOT a valid model name'
+            raise CommentBadRequest(self.error)
+        except ValueError:
+            self.error = f'model id must be an integer, {self.model_id} is NOT'
+            raise CommentBadRequest(self.error)
```

Comment:
Too many if else doesn't always give great readability to the code. This was refactored in an earlier commit, maybe you missed it.

```python

    def _set_error_for_model_name(self, val):
        if not val:
            self.error = "model name must be provided"
           raise CommentBadRequest(self.error)

    def _set_error_for_model_id(self, val):
        if not val:
            self.error = "model id must be provided"
            raise CommentBadRequest(self.error)

    def _set_error_for_app_name(self, val):
        if not val:
            self.error = 'app name must be provided'
            raise CommentBadRequest(self.error)

        if not ContentType.objects.filter(app_label=val).exists():
            self.error = f'{val} is NOT a valid app name'
            raise CommentBadRequest(self.error)

    def _set_error(self):
        try:
            model_name = self.model_name.lower()
            ct = ContentType.objects.get(model=model_name).model_class()
            model_class = ct.objects.filter(id=self.model_id)
            if not model_class.exists() and model_class.count() != 1:
                self.error = f"{self.model_id} is NOT a valid model id for the model {self.model_name}"
        except ContentType.DoesNotExist:
            self.error = f"{self.model_name} is NOT a valid model name"
        except ValueError:
            self.error = f"model id must be an integer, {self.model_id} is NOT"

    def validate(self, request):
         super().validate(request)
         self._set_error_for_model_name(self.model_name)
         self._set_error_for_model_id(self.model_id)
         self._set_error_for_app_name(self.app_name)
         self._set_error()
```